### ISSUE TITLE
opentelemetry-resourcedetector-gcp update google-auth dependency pinning

### ISSUE THREAD

**Issue by frehov** [GoogleCloudPlatform/opentelemetry-operations-python#165]
google-auth has been released in version 2.0.0 and the breaking change is the removal of support for python 2.7.
As far as I can see, the target for opentelemetry-resourcedetector-gcp is python 3 with python 3.6 as the minimum requirement.

Would you allow for opening up the dependency pinning to account for the 2.x release series of r google-auth?

**Comment by aabmass** [comment]
Don't have time to tackle this right now, but it should be easy to fix.

If we are extending beyond a single major version, it would be great to have testing for both.

**Comment by bourgeoisor** [comment]
I'm having a semi-related issue relating to `google-auth` where I can't upgrade Flask from v1 to v2.

If I take it one dep at a time trying to do minimal changes, `flask` requires a bump of `opentelemetry-instrumentation-flask` which requires a bump of `opentelemetry-api` which in turn requires all `opentelemetry`-related deps to be bumped to 1.x, which requires `google-cloud`-related deps to be bumped, which means bumping `google-auth` to v2.

This chain of bumping deps all work great except for this:
```
There are incompatible versions in the resolved dependencies:
  google-auth~=1.22 (from opentelemetry-resourcedetector-gcp==1.1.0a0->-r requirements.in (line 27))
  google-auth<3.0dev,>=1.25.0 (from google-api-core==2.7.1->-r requirements.in (line 11))
  google-auth<3.0dev,>=1.25.0 (from google-cloud-core==2.2.3->-r requirements.in (line 13))
```

In short, both `google-api-core` and `google-cloud-core` rely on `google-auth` being _at least_ `1.25.0` but the latest version of `opentelemetry-resourcedetector-gcp` asks for `~1.22` so they conflict.

Is there a version of the GCP opentelemetry deps that works with either `google-auth` 1.25 or v2? How should I proceed?

**Comment by aabmass** [comment]
@frehov you are right that these packages only support python3.6+. Would upgrading the dep to `google-auth ~= 2.0` (and no longer supporting 1.x) fix this for both of you?

**Comment by bourgeoisor** [comment]
The latest releases of `google-api-core` and `google-cloud-core` both require ` google-auth<3.0dev,>=1.25.0`, so it looks like anything in the `2.x` range should be good!

**Comment by aabmass** [comment]
This was fixed in #176. David realized that the library isn't actually even using google-auth so there's no dependency anymore
